controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -411,12 +493,30 @@ func (sp *servicePublisher) newPortPublisher(srcPort Port, hostname string) *por
 		metrics:    endpointsVecs.newEndpointsMetrics(sp.metricsLabels(srcPort, hostname)),
 	}
 
-	endpoints, err := sp.k8sAPI.Endpoint().Lister().Endpoints(sp.id.Namespace).Get(sp.id.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
-		sp.log.Errorf("error getting endpoints: %s", err)
-	}
-	if err == nil {
-		port.updateEndpoints(endpoints)
+	err = pkgK8s.EndpointSliceAccess(sp.k8sAPI.Client)
+	if err != nil {
+		sp.log.Debugf("No EndpointSlice access, using endpoints:%v", err)
+		endpoints, err := sp.k8sAPI.Endpoint().Lister().Endpoints(sp.id.Namespace).Get(sp.id.Name)
+		if err != nil && !apierrors.IsNotFound(err) {
+			sp.log.Errorf("error getting endpoints: %s", err)
+		}
+		if err == nil {
+			port.updateEndpoints(endpoints)
+		}
+	} else {
```

Comment:
While I agree that we should only have 1 informer of type Endpoints or 1 of type EndpointSlice, since EndpointSlice will be on by default, perhaps it's a good idea we always update and populate new publishers with Endpoints?

We can eliminate an extra step in the processing, the whole port gets updated/instantiated at once, and makes it easier to understand. We are not using informers here just the client, thoughts?

Reply:
That sounds good. The topology data slipped my mind, will opt for consistency and make all relevant changes. Thank you!